GraphQLFieldConfig prevents arguments from being typed

Using a non-optional type for field arguments is not possible with current TypeScript typings. Here's a small reproduction:

```typescript
import {
  GraphQLNonNull,
  GraphQLObjectType,
  GraphQLSchema,
  GraphQLString,
} from "graphql";

const queryType = new GraphQLObjectType({
  name: "Query",
  fields: {
    hello: {
      type: new GraphQLNonNull(GraphQLString),
      args: { message: { type: new GraphQLNonNull(GraphQLString) } },
      resolve(rootValue, args: { message: string }) {
        return `Hello, ${args.message}!`;
      },
    },
  },
});

export const schema = new GraphQLSchema({ query: queryType });
```
```
src/schema.ts:14:7 - error TS2322: Type '(rootValue: any, args: { message: string; }) => string' is not assignable to type 'GraphQLFieldResolver<any, any, { [argName: string]: any; }>'.
  Types of parameters 'args' and 'args' are incompatible.
    Property 'message' is missing in type '{ [argName: string]: any; }' but required in type '{ message: string; }'.

14       resolve(rootValue, args: { message: string }) {
         ~~~~~~~

  src/schema.ts:14:34
    14       resolve(rootValue, args: { message: string }) {
                                        ~~~~~~~
    'message' is declared here.
  node_modules/graphql/type/definition.d.ts:519:3
    519   resolve?: GraphQLFieldResolver<TSource, TContext, TArgs>;
          ~~~~~~~
    The expected type comes from property 'resolve' which is declared here on type 'GraphQLFieldConfig<any, any, { [argName: string]: any; }>'


Found 1 error.
```

If the type of args is changed to `{ message?: string }` the problem goes away. But this isn't the correct type since that input field is non-null.

--- a/integrationTests/ts/basic-test.ts
+++ b/integrationTests/ts/basic-test.ts
@@ -9,10 +9,13 @@ const queryType: GraphQLObjectType = new GraphQLObjectType({
     sayHi: {
       type: GraphQLString,
       args: {
-        who: { type: GraphQLString },
+        who: {
+          type: GraphQLString,
+          defaultValue: 'World',
+        },
       },
-      resolve(_root, args) {
-        return 'Hello ' + (args.who ?? 'World');
+      resolve(_root, args: { who: string }) {
+        return 'Hello ' + args.who;
       },
     },
   }),

--- a/src/type/definition.ts
+++ b/src/type/definition.ts
@@ -962,7 +962,7 @@ export type GraphQLIsTypeOfFn<TSource, TContext> = (
 export type GraphQLFieldResolver<
   TSource,
   TContext,
-  TArgs = { [argument: string]: any },
+  TArgs = any,
   TResult = unknown,
 > = (
   source: TSource,
@@ -996,19 +996,11 @@ export interface GraphQLResolveInfo {
  * We've provided these template arguments because this is an open type and
  * you may find them useful.
  */
-export interface GraphQLFieldExtensions<
-  _TSource,
-  _TContext,
-  _TArgs = { [argName: string]: any },
-> {
+export interface GraphQLFieldExtensions<_TSource, _TContext, _TArgs = any> {
   [attributeName: string]: unknown;
 }
 
-export interface GraphQLFieldConfig<
-  TSource,
-  TContext,
-  TArgs = { [argument: string]: any },
-> {
+export interface GraphQLFieldConfig<TSource, TContext, TArgs = any> {
   description?: Maybe<string>;
   type: GraphQLOutputType;
   args?: GraphQLFieldConfigArgumentMap;
@@ -1049,11 +1041,7 @@ export type GraphQLFieldConfigMap<TSource, TContext> = ObjMap<
   GraphQLFieldConfig<TSource, TContext>
 >;
 
-export interface GraphQLField<
-  TSource,
-  TContext,
-  TArgs = { [argument: string]: any },
-> {
+export interface GraphQLField<TSource, TContext, TArgs = any> {
   name: string;
   description: Maybe<string>;
   type: GraphQLOutputType;

